src/navigator/Navigator.tsx

```diff
@@ -634,6 +639,12 @@ export function MainStackScreen() {
       initialRoute = Screens.Welcome
     } else if (!account) {
       initialRoute = choseToRestoreAccount ? Screens.ImportWallet : Screens.Welcome
+    } else if (
```

Comment:
At first glance it feels like this if/else chain is ripe for refactor, for example doing a `lastOnboardingScreenSeen` variable in redux, integrating it into steps.ts and deciding which screen to load after a bailout based on that. 

However, that would mean a few real product changes, like taking the user directly to name screen, pin screen, biometrics screen which we may or may not want to do. 

I'm sure you thought about this, but I'm okay with your current implementation b/c all the exceptions to the rules right now I think would defeat the point of a refactor.

Reply:
yeah, we should eventually do this, but outside the scope of the current ticket as we discussed in the sync. Will create a follow up ticket. One idea to retain current behavior is to have some onboarding milestones, that way we don't land on certain screens we currently don't.